From 7836dc2dfeee4b7b1ea0b25afd737e372a4a6250 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@freescale.com>
Date: Tue, 28 Apr 2015 14:41:36 +0800
Subject: [PATCH 1407/1594] MLK-10718 usb: chipidea: udc: disable BSV irq when
 otg from undefined to a_idle

commit bf6a1cdaef68125cc10240f174019e2646ebaa24 from
git://git.freescale.com/imx/linux-2.6-imx.git

After device power up, insert usb OTG ID cable, otg fsm from undefined state
to a_idle, the BSV cannot be disabled, this will result BSV irq generated but
which is actually only for B device.

Signed-off-by: Li Jun <jun.li@freescale.com>
---
 drivers/usb/chipidea/udc.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/usb/chipidea/udc.c b/drivers/usb/chipidea/udc.c
index 4eb6f96..3817398 100644
--- a/drivers/usb/chipidea/udc.c
+++ b/drivers/usb/chipidea/udc.c
@@ -2009,7 +2009,8 @@ static void udc_id_switch_for_host(struct ci_hdrc *ci)
 	 * (in otg fsm mode, means B_IDLE->A_IDLE) due to ID change.
 	 */
 	if (!ci_otg_is_fsm_mode(ci) ||
-			ci->fsm.otg->state == OTG_STATE_B_IDLE)
+			ci->fsm.otg->state == OTG_STATE_B_IDLE ||
+			ci->fsm.otg->state == OTG_STATE_UNDEFINED)
 		hw_write_otgsc(ci, OTGSC_BSVIE | OTGSC_BSVIS, OTGSC_BSVIS);
 }
 
-- 
1.7.5.4

